test/e2e/sync_plugin_test.go

```diff
@@ -712,52 +713,73 @@ func TestSecretCredentialSyncAfterStartup(t *testing.T) {
+	cm := newPodTemplateConfigMap(cmName, podTemplateName, labels)
+	cm, err := kubeClient.CoreV1().ConfigMaps(ta.ns).Create(context.Background(), cm, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("error creating pod template cm: %s", err.Error())
+	}
+	podTemplateTest(podTemplateName, ta)
+}
 
-	cm := &corev1.ConfigMap{Data: map[string]string{"jenkins-agent": `
-      <org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
-        <inheritFrom></inheritFrom>
-        <name>jenkins-agent</name>
-        <instanceCap>2147483647</instanceCap>
-        <idleMinutes>0</idleMinutes>
-        <label>jenkins-agent</label>
-        <serviceAccount>jenkins</serviceAccount>
-        <nodeSelector></nodeSelector>
-        <volumes/>
-        <containers>
-          <org.csanchez.jenkins.plugins.kubernetes.ContainerTemplate>
-            <name>jnlp</name>
-            <image>image-registry.openshift-image-registry.svc:5000/openshift/jenkins-agent-maven:latest</image>
-            <privileged>false</privileged>
-            <alwaysPullImage>false</alwaysPullImage>
-            <workingDir>/tmp</workingDir>
-            <command></command>
-            <args>${computer.jnlpmac} ${computer.name}</args>
-            <ttyEnabled>false</ttyEnabled>
-            <resourceRequestCpu></resourceRequestCpu>
-            <resourceRequestMemory></resourceRequestMemory>
-            <resourceLimitCpu></resourceLimitCpu>
-            <resourceLimitMemory></resourceLimitMemory>
-            <envVars/>
-          </org.csanchez.jenkins.plugins.kubernetes.ContainerTemplate>
-        </containers>
-        <envVars/>
-        <annotations/>
-        <imagePullSecrets/>
-        <nodeProperties/>
-      </org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
-`}}
-	cm.Labels = map[string]string{"role": "jenkins-slave"}
-	cm.Name = "config-map-with-podtemplate"
-
+func TestConfigMapLegacyPodTemplate(t *testing.T) {
+	ta := setupThroughJenkinsLaunch(t, nil)
+	defer projectClient.ProjectV1().Projects().Delete(context.Background(), ta.ns, metav1.DeleteOptions{})
+	labels := map[string]string{"role": "jenkins-slave"}
+	cmName := "config-map-with-legacy-podtemplate"
+	podTemplateName := "jenkins-slave"
+	cm := newPodTemplateConfigMap(cmName, podTemplateName, labels)
 	cm, err := kubeClient.CoreV1().ConfigMaps(ta.ns).Create(context.Background(), cm, metav1.CreateOptions{})
 	if err != nil {
 		t.Fatalf("error creating pod template cm: %s", err.Error())
 	}
+	podTemplateTest(podTemplateName, ta)
+}
 
-	podTemplateTest("sync-plugin-configmap-pod-template", ta)
+func newPodTemplateConfigMap(configMapName string, podTemplateName string, templateLabels map[string]string) *corev1.ConfigMap {
+	templateDefinition := `
+	      <org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
+	        <inheritFrom></inheritFrom>
+	        <name>POD_TEMPLATE_NAME</name>
+	        <instanceCap>2147483647</instanceCap>
+	        <idleMinutes>0</idleMinutes>
+	        <label>jenkins-agent</label>
```

Comment:
```suggestion
	        <label>POD_TEMPLATE_NAME</label>
```

Reply:
nice catch @gabemontero ; I was thinking that only the name was important here and I let this label as is for that. But, when the test broke  I forgot to return to it and fix it.